src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -637,6 +654,99 @@ public bool IsTypeDesc
 
             return (MethodTable*)m_asTAddr;
         }
+
+        /// <summary>
+        /// Gets a <see cref="MethodTable"/> pointer for the current type handle.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public MethodTable* GetMethodTable()
+        {
+            if (IsTypeDesc)
+            {
+                return AsTypeDesc()->GetMethodTable();
+            }
+
+            return (MethodTable*)m_asTAddr;
+        }
+    }
+
+    internal unsafe struct TypeDesc
+    {
+        /// <summary>
+        /// <para>
+        /// The low-order 8 bits of this flag are used to store the <see cref="CorElementType"/>,
+        /// which discriminates what kind of <see cref="TypeDesc"/> this is.
+        /// </para>
+        /// <para>The remaining bits are available for flags.</para>
+        /// </summary>
+        private readonly uint TypeAndFlags;
+
+        /// <summary>
+        /// Gets a <see cref="MethodTable"/> pointer for the current type desc.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public MethodTable* GetMethodTable()
+        {
+            // The native version here has a check for IsGenericVariable, which causes this method to return null.
+            // This path is never taken by TypedReference, as it always assumes the returned MethodTable* pointer
+            // is not null. If adding code that needs this path to be available as well, it needs to be ported too.
+            //
+            // if (IsGenericVariable)
+            // {
+            //     return null;
+            // }
+
+            if (GetInternalCorElementType() == CorElementType.ELEMENT_TYPE_FNPTR)
```

Comment:
We need to add a test for this case.

Reply:
Added a test for pointers and one for function pointers in 871e46c83a7e80a09ed4e207e5f32e9e63d0ddd1 🙂